test/e2e/json-rpc/wallet_addEthereumChain.spec.ts

```diff
@@ -434,4 +434,69 @@ describe('Add Ethereum Chain', function () {
       );
     });
   });
+  describe('the dapp is already permitted to use the chain being added, and the dapp is on the same chain as the chain being added, but the rpcEndpoint being proposed does not match any existing rpcEndpoints for the chain', () => {
+    it('prompts to add the rpc endpoint to the chain networkConfiguration and set it as the default', async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withNetworkControllerDoubleGanache()
+            .withPermissionControllerConnectedToTestDappWithChains(['0x539'])
+            .build(),
+          ganacheOptions: {
+            ...defaultGanacheOptions,
+            concurrent: [{ port: 8546, chainId: '0x539' }],
+          },
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver }: { driver: Driver }) => {
+          await unlockWallet(driver);
+          await openDapp(driver);
+
+          const beforePermittedChains = await getPermittedChains(driver);
+          assert.deepEqual(beforePermittedChains, ['0x539']);
+
+          await driver.findElement({ css: '#chainId', text: '0x539' });
+
+          const addEthereumChainRequest = JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'wallet_addEthereumChain',
+            params: [
+              {
+                chainId: '0x539',
+                chainName: 'Alternative localhost chain 0x539',
+                nativeCurrency: {
+                  name: '',
+                  symbol: 'ETH',
+                  decimals: 18,
+                },
+                // this does not match what already exists in the NetworkController as an endpoint for this chain
+                rpcUrls: ['http://127.0.0.1:8546'],
```

Comment:
you can use localhost here i think and keep it on 8545

Reply:
https://github.com/MetaMask/metamask-extension/pull/29885/commits/e087e519199e717e63778be973a69fac37c513d5